services/web/server/src/simcore_service_webserver/catalog_settings.py

```diff
@@ -3,28 +3,43 @@
     - config-file schema
     - settings
 """
-from typing import Dict
+from functools import cached_property
 
 from aiohttp import web
 from models_library.basic_types import PortInt, VersionTag
-from pydantic import BaseSettings, Field
+from settings_library.base import BaseCustomSettings
+from settings_library.utils_service import DEFAULT_FASTAPI_PORT, MixinServiceSettings
 
 from .catalog_config import get_config
 
 
-class CatalogSettings(BaseSettings):
-    enabled: bool = True
-    host: str = "catalog"
-    port: PortInt = 8000
-    vtag: VersionTag = Field(
-        "v0", alias="version", description="Catalog service API's version tag"
-    )
+class CatalogSettings(BaseCustomSettings, MixinServiceSettings):
+    CATALOG_HOST: str = "catalog"
+    CATALOG_PORT: PortInt = DEFAULT_FASTAPI_PORT
+    CATALOG_VTAG: VersionTag = "v0"
 
-    class Config:
-        prefix = "CATALOG_"
+    @cached_property
+    def base_url(self) -> str:
+        return self._build_api_base_url(prefix="CATALOG")
 
+    @cached_property
+    def origin(self) -> str:
```

Comment:
is there a rationale when you need the origin?

Reply:
URL conventions (based on https://yarl.readthedocs.io/en/latest/api.html)
    
         http://user:pass@service.com:8042/v0/resource?name=ferret#nose
         \__/   \__/ \__/ \_________/ \__/\__________/ \_________/ \__/
         |      |    |        |       |      |           |        |
         scheme  user password host    port   path       query   fragment
    
     origin    -> http://example.com
     api_base  -> http://example.com:8042/v0
    

It was used somewhere in the code and I added a ``MixinServiceSettings`` to keep these names consistent with its construction.